numba_dppy/utils/array_utils.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import dpctl
+import dpctl.memory as dpctl_mem
+
+DEVICE_ARRAY_ATTR = "__sycl_usm_array_interface__"
+
+
+def _is_device_accessible_array(arr):
+    """
+    Function to determine if a given array is SYCL device accessible.
+
+    Args:
+        arr: array_like object.
+
+    Retruns:
+        bool: True if array is device accessible, False otherwise.
+    """
+    if hasattr(arr, DEVICE_ARRAY_ATTR):
```

Comment:
IMO just checking if the attribute exists is not enough. Look at this Pytorch test case https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/test/test_numba_integration.py#L84.

We should check the contents of the dictionary and the version of the interface.